### DISCUSSION TITLE
Modelling node faults when using Ieee802154NarrowbandInterface

### DISCUSSION THREAD

**Discussion by n-jay** [omnetpp/omnetpp#1026] [Q&A] [answered]
Hi all,
I've been running a WSN simulation and trying to model node faults using the **ScenarioManager**. 
The sensor network uses the `Ieee802154NarrowbandInterface` module for the wlan type.

The nodes in the network are using a hierarchical routing protocol I've written. 
The node module sub-components look like this in the NED visual view:
![Selection_613](https://github.com/omnetpp/omnetpp/assets/24616975/32b4f210-cf0f-4064-aa11-12ae41089e83)
The arrow indicates where my routing protocol module resides in the stack.

The nodes are linked to the modules by simply referencing them using the type parameter in the .ini file.
```
*.baseStation.typename = "LEACHbs"
*.host*.typename = "LEACHnode"
``` 

My scenario.xml file looks like this:
![Selection_614](https://github.com/omnetpp/omnetpp/assets/24616975/74845bc1-0122-4283-8c71-3f9b7b120653)
The node numbers and time slots are completely random.

However, whenever I try to simulate hard faults in the network, I'm faced with an error of the following type:
```
<!> Self message 'timer-backoff' received when Ieee802154NarrowbandMac is down 
-- in module (inet::Ieee802154Mac) SimpleNet.host[11].wlan[0].mac
``` 
I presume the "timer-backoff" message is from the CSMA module associated with `Ieee802154NarrowbandMac` and it's extremely annoying to keep changing my XML to get it running properly every time it crashes.

Are there any suggestions on how I can resolve this?

**Comment by n-jay** [comment]
@rhornig appreciate some thoughts on this.

**Reply by rhornig** [reply]
:shrug: sadly I have no experience with this. :disappointed:

**Reply by n-jay** [reply]
No worries 😄 
Wonder if @levy would have an idea

**Comment by n-jay** [comment, marked as answer]
Hi all,
the issue turned out to be that the version of INET I'm using (4.2.5) does not have the updated IEEE 802.15.4 module which is missing the relevant `handleStopOperation` and `handleCrashOperation` methods.
These are present in the latest version of that file:
https://github.com/inet-framework/inet/blob/b02f7bedb7b7168640de465d476190af946ce6ac/src/inet/linklayer/ieee802154/Ieee802154Mac.cc#L923

The resolution was to add the methods manually and rerun the simulation.
